packages/react-centra-checkout/src/PaymentForm/index.tsx

```diff
@@ -0,0 +1,84 @@
+import * as Centra from '@noaignite/centra-types'
+import * as React from 'react'
+import { useCentra } from '../Context'
+import PaymentFormEmbed from './partials/PaymentFormEmbed'
+
+interface PaymentFormProps {
+  paymentReturnPage: string
+  paymentFailedPage: string
+  onSuccess?(paymentResult: Centra.PaymentResponse): void
+  onError?(error: Record<string, string>): void
+}
+
+/** This component handles rendering of payment widgets such as Klarna Checkout and Adyen drop-in, if you submit payments yourself directly,
+you should simply call the submitPayment method of the context instead */
+function PaymentForm(props: PaymentFormProps): React.ReactElement | null {
+  const { paymentReturnPage, paymentFailedPage, onError, onSuccess } = props
+
+  const [paymentResult, setPaymentResult] = React.useState<Centra.PaymentResponse | null>(null)
+  const [formHtml, setFormHtml] = React.useState<string | null>(null)
+
+  // get selection
+  const { selection, paymentMethods, submitPayment } = useCentra()
+  // get payment method
+  const paymentMethodId = selection?.paymentMethod
+  const paymentMethod = paymentMethods?.find((p) => p.paymentMethod === paymentMethodId)
+
+  // POST to /payment if payment method is initiate only (method that doesn't need address) or if the payment method provides it's own address
+  React.useEffect(() => {
+    if (
+      selection &&
+      paymentMethod &&
+      (paymentMethod.providesCustomerAddressAfterPayment || paymentMethod.supportsInitiateOnly)
+    ) {
+      const { address, shippingAddress } = selection
+
+      submitPayment?.({
```

Comment:
How come this is needed as optional chaining @alexanderflink ?

Reply:
![image](https://user-images.githubusercontent.com/83586699/150497467-8854496a-94b8-4b2e-a066-5a213e1bdce8.png)

Reply:
This is sort of typescript specific, but it is because none of the properties of the useCentra are actually guaranteed to exist. For example, if you try to access `submitPayment` but haven't wrapped your component in the `CentraProvider`. I think it's nice though, this ensures your app won't crash if you miss this.

Reply:
Awesome, thanks!

Reply:
![image](https://user-images.githubusercontent.com/54660936/150499509-7831fe83-f4ef-4cda-843e-eab0cc8a6994.png)